pkg/secrets/providers.go

```diff
@@ -22,7 +22,7 @@ func NewProviders() *Providers {
 
 // Register registers a provider with associated spec. This
 // is expected to happen during app startup.
-func (ps *Providers) Register(sp SecretStoreProvider, spec *secrets.ProviderSpec) {
+func (ps *Providers) Register(sp SecretStoreFactory, spec *secrets.ProviderSpec) {
```

Comment:
Please consider renaming`sp` to `sf` or `factory`, as you have renamed the structure. Additionally, I noticed that `SecretStoreFactory` is a struct while `ProviderSpec` is a pointer. Is this intentional or should they have the same type? Please clarify.

Reply:
Just rename `sp` to `ssf`. Actually `SecretStoreFactory` is an interface, and this is intentional.